[12.0][MIG] l10n_be_vat_reports: VAT Listing and VAT Intra

robinkeunen opened the pull request:
This migration is based on the v8 odoo standards reports for _VAT Listing_ and _VAT Intra_. The work was started by @andreparames at #57, improved at #74 for v10 and #75 for v9.

This time, I tried to include original v8 code as @sbidoul recommended (see 4e5eccc and 9ca1106).

Since reports have changed in v12, I refactored most of the code to avoid passing data to the report templates: everything is on the model.

andreparames committed:
Reports ported: VAT Listing and VAT Intra

andreparames committed:
[FIX] Correctly calculate turnover and vat_amount

This commit fixes two problems with the calculation:

- The turnover tax was being fetched from the wrong field
- Move lines with taxes that had more than one tag where being
summed more than once
Use -balance for turnover and tax

andreparames committed:
[FIX] vat reports: Create XML records in current module

They were being created in l10n_be

andreparames committed:
vat reports: add test

andreparames committed:
vat reports: improve backward compatibility

andreparames committed:
vat reports: upgrade sequence code

andreparames committed:
vat reports: fixing menu references

andreparames committed:
vat reports: fix UI problems in VAT intra report

andreparames committed:
vat reports: improved intra report

andreparames committed:
vat reports: fix encoding & remove pointless code

andreparames committed:
vat reports: small corrections

robinkeunen committed:
[ADD] l10n_be_vat_reports: copy code from
odoo/odoo/tree/8.0/addons/l10n_be

https://github.com/odoo/odoo/tree/8.0/addons/l10n_be

robinkeunen committed:
[REF] l10n_be_vat_reports: blacken

robinkeunen committed:
[ADD] l10n_be_vat_reports: credits to Odoo SA

robinkeunen committed:
[MIG] l10n_be_vat_reports: 11.0

robinkeunen committed:
[MIG] l10n_be_vat_reports: 12.0

robinkeunen committed:
[MIG] l10n_be_vat_reports: Annual Listing of VAT-Subjected

robinkeunen committed:
[MIG] l10n_be_vat_reports: Partner VAT Intra

robinkeunen committed:
[IMP] l10n_be_vat_reports: use the same save window

houssine78 commented:
Hi @sbidoul do you think that this PR can be merged ?

houssine78 committed:
[FIX] add missing where clause

houssine78 committed:
[IMP] add Odoo SA to the authors

remytms reviewed (approved):
LGTM

vvrossem commented on l10n_be_vat_reports/views/report_vatintraprint.xml:
```diff
@@ -0,0 +1,53 @@
+                                        <td><span t-field="l.vat"/></td>
+                                        <td><span t-field="l.code"/></td>
+                                        <td><span t-field="l.amount"/> <span t-field="res_company.currency_id"/></td>
+<!--                                        <td><span t-field="l.amount" t-field-options='{"widget": "monetary", "display_currency": "res_company.currency_id"}'/></td>-->
```
should be removed ?

vvrossem commented on l10n_be_vat_reports/views/report_vatintraprint.xml:
```diff
@@ -0,0 +1,53 @@
+                            </table>
+
+                            <p><strong>Total: </strong><span t-field="o.amount_total"/> <span t-field="res_company.currency_id"/></p>
+<!--                            <p><strong>Total: </strong><span t-field="o.amount_total" t-field-options='{"widget": "monetary", "display_currency": "res_company.currency_id"}'/></p>-->
```
should be removed?

vvrossem commented on l10n_be_vat_reports/wizard/l10n_be_partner_vat_listing.py:
```diff
@@ -0,0 +1,395 @@
+
+    @api.multi
+    def create_xml(self):
+        # why is the sequence number skipped here?
```
should be solved or removed?

vvrossem commented on l10n_be_vat_reports/wizard/l10n_be_vat_intra_view.xml:
```diff
@@ -0,0 +1,116 @@
+                        <button name="get_partners" string="View Customers" type="object"  class="oe_highlight"  />
+                        or
+                        <button string="Cancel" class="oe_link" special="cancel" />
+<!--                        <button name="create_xml" string="Create _XML" type="object" class="oe_highlight"/>-->
```
should be removed ?

vvrossem reviewed (commented):
LGTM, but some commented lines need to be checked

houssine78 committed:
[IMP] switch to the standard way to display currency field

[IMP] remove commented code

enricostano commented:
Hi @robinkeunen @houssine78 what's missing to get merged this? Can we help in any way?

robinkeunen commented:
We were waiting for a review from @sbidoul but this module should probably be recoded in a cleaner way like in [this PR](https://github.com/OCA/l10n-belgium/pull/99)

houssine78 commented:
@robinkeunen this PR can be merged as it is as it a porting from previous version. At least we should take into account the comment from @vvrossem by removing commented lines. We use this PR in production for more than two years now. refactoring on this version would probably means never get it merged. Let's keep pragmatic.

houssine78 commented:
Hi @sbidoul could you have a quick look on this PR ? all the comments have been taken into account. 

thanks !

sbidoul commented on l10n_be_vat_reports/__manifest__.py:
```diff
@@ -0,0 +1,42 @@
+##############################################################################
+#
+#     This file is part of l10n_be_mis_reports,
+#     an Odoo module.
```
This comment is incorrect. BTW, you could simplify the copyright headers using the OCA template and also add Odoo and CoopIT Easy ?

sbidoul commented:
I folks, sorry for the delay. I'd just be extra careful with the copyright attribution in all files, to avoid all possible trouble with Odoo.

The first commit is the same as Odoo 8.0, right ?

robinkeunen commented:
Hi @sbidoul,

- I rebased for a cleaner history
- the first commit (6445398fbcb921c6f8c03f15c389b5b1774c0123) is the code from V8
- I reformated the comment section in `__manifest__` and other `.py` or `.xml` files but I left _tiny sprl_ as it is the company referred to in all v8 code
- On the other hand, I added Odoo SA under the author key of the manifest. I think that's enough :-)

sbidoul commented:
Great thanks! Let's merge this at last.

/ocabot merge patch
